stubs/tensorflow/tensorflow/nn.pyi

```diff
@@ -0,0 +1,195 @@
+    data_format: Literal["NHWC", "NCHW"] = "NHWC",
+    dilations: int | Sequence[int] | None = None,
+    name: str | None = None,
+) -> Tensor: ...
+def conv3d(
+    input: TensorOrArray,
+    filters: TensorOrArray,
+    strides: int | Sequence[int],
+    padding: Literal["VALID", "SAME"],
+    data_format: Literal["NDHWC", "NCDHW"] = "NDHWC",
+    dilations: int | Sequence[int] | None = None,
+    name: str | None = None,
+) -> Tensor: ...
+def conv3d_transpose(
+    input: TensorOrArray,
+    filters: TensorOrArray,
+    output_shape: TensorOrArray,
+    strides: int | Sequence[int],
+    padding: Literal["VALID", "SAME"] = "SAME",
+    data_format: Literal["NDHWC", "NCDHW"] = "NDHWC",
+    dilations: int | Sequence[int] | None = None,
+    name: str | None = None,
+) -> Tensor: ...
+def conv_transpose(
+    input: TensorOrArray,
+    filters: TensorOrArray,
+    output_shape: TensorOrArray,
+    strides: int | Sequence[int],
+    padding: Literal["VALID", "SAME"] = "SAME",
+    data_format: str | None = None,
+    dilations: int | Sequence[int] | None = None,
+    name: str | None = None,
+) -> Tensor: ...
+def convolution(
+    input: TensorOrArray,
+    filters: TensorOrArray,
+    strides: int | Sequence[int] | None = None,
+    padding: Literal["VALID", "SAME"] = "VALID",
+    data_format: Literal["NC", "NWC", "NCW", "NHWC", "NCHW", "NDHWC", "NCDHW"] | None = None,
+    dilations: int | Sequence[int] | None = None,
+    name: str | None = None,
+) -> Tensor: ...
+def crelu(features: TensorOrArray, axis: int = -1, name: str | None = None) -> Tensor: ...
+def ctc_beam_search_decoder(
+    inputs: TensorOrArray, sequence_length: TensorOrArray | Sequence[int], beam_width: int = 100, top_paths: int = 1
+) -> tuple[list[SparseTensor], Tensor]: ...
+def ctc_greedy_decoder(
+    inputs: Tensor, sequence_length: Tensor | Sequence[int], merge_repeated: bool = True, blank_index: int | None = None
+) -> tuple[list[SparseTensor], Tensor]: ...
+def ctc_loss(
+    labels: Tensor,
+    logits: Tensor,
+    label_length: Tensor,
+    logit_length: Tensor,
+    logits_time_major: bool = True,
+    unique: int | None = None,
+    blank_index: int | None = None,
+    name: str | None = None,
+) -> Tensor: ...
+def ctc_unique_labels(labels: Tensor, name: str | None = None) -> tuple[Tensor, Tensor]: ...
+@overload
+def embedding_lookup(
+    params: TensorCompatible, ids: TensorCompatible, max_norm: float | None = None, name: str | None = None
+) -> Tensor: ...
+@overload
+def embedding_lookup(
+    params: TensorCompatible, ids: RaggedTensor, max_norm: float | None = None, name: str | None = None
+) -> RaggedTensor: ...
+def leaky_relu(features: TensorCompatible, alpha: float = 0.2, name: str | None = None) -> Tensor: ...
+def log_poisson_loss(
+    targets: TensorCompatible, log_input: TensorCompatible, compute_full_loss: bool = False, name: str | None = None
+) -> Tensor: ...
+
+# tf.nn.moments's shit is not used in the current implementation.
```

Comment:
```suggestion
# tf.nn.moments's shift is not used in the current implementation.
```
Not sure if this is profanity or a typo. In either case, let's keep it out of the repo :)